security/tests/agent/README.md

```diff
@@ -0,0 +1,30 @@
+# Testing Citadel Agent
+
+## Getting Started
+
+Starting the Citadel Agent first
+
+```bash
+go build  -o agent \
+  ${GOPATH}/src/istio.io/istio/security/cmd/node_agent_k8s/main.go
+
+export AGENT_UDS_PATH=$(mktemp /tmp/citadel-agent.XXXX)
+echo "Citadel Agent UDS Path ${AGENT_UDS_PATH}"
+CA_PROVIDER='VaultCA' CA_ADDR="https://34.83.129.211:8200" ./agent \
+  --workloadUDSPath=${AGENT_UDS_PATH}
+```
```

Comment:
We may need a test CA backend that exposes interface to manipulate its behavior. E.g. (1) timeout/broken connection for CSR request (2) wrong format in cert response (3) reject CSR due to token validation failure. etc.

Reply:
alternatively we can add a testing ca provider just compiled inside citadel agent. no outbound request to ca backend at all. everything is done in memory.

Reply:
Good point.